src/main/java/training/supportbank/Account.java

```diff
@@ -0,0 +1,58 @@
+package training.supportbank;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Account {
+
+    private final String name;
+    private List<Transaction> transactions;
+    private Float balance;
+
+    public Account(String name) {
+        this.name = name;
+        this.balance = Float.valueOf(0);
+        this.transactions = new ArrayList<>();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Float getBalance() {
+        return balance;
+    }
+
+    public List<Transaction> getTransactions() {
+        return transactions;
+    }
+
+    public void processTransaction(Transaction transaction) {
+        if (name.equals(transaction.getFrom())) {
+            debit(transaction.getAmount());
+        } else if (name.equals(transaction.getTo())) {
+            credit(transaction.getAmount());
+        }
+        transactions.add(transaction);
+    }
+
+    @Override
+    public boolean equals(Object other) {

```

Comment:
There's a subtlety with `equals` that I didn't mention on Friday - if you override `equals`, you have to remember to override `hashCode`. That's because both methods are used if you put the objects into a Set (or use them as a the keys in a Map) - you might end up with duplicates in the set because their hashCodes are different even though `equals` thinks they're the same.

Writing hashCodes is tedious, so this is where I'd suggest using a library to make it easier. [EqualsBuilder](https://commons.apache.org/proper/commons-lang/javadocs/api-release/org/apache/commons/lang3/builder/EqualsBuilder.html) and [HashCodeBuilder](https://commons.apache.org/proper/commons-lang/javadocs/api-release/org/apache/commons/lang3/builder/HashCodeBuilder.html) in [Apache Commons Lang](https://commons.apache.org/proper/commons-lang/index.html) are a good choice.

It's a pain to have to implement `equals` and `hashCode`, and a pain to have to add an extra util library to deal with them. I think there are plans to add value types to a future version of Java to save you from having to do all of this.